File: 백준/Bronze/2884. 알람 시계/알람 시계.py
h, m = map(int, input().split())

mm = m - 45

if mm < 0:
    h -= 1
    if h < 0:
        h += 24
    m = 60 + mm

elif mm > 0:
    m = m - 45

elif mm == 0:
    m -= 45

elif h < 0:
    h += 24

print(h, m)

File: 백준/Bronze/2884. 알람 시계/알람 시계.kt
fun main() {

    var input = readln().split(" ")

    var H = input[0].toInt()
    var M = input[1].toInt()

    if (M - 45 < 0) {
        M = 60 + M - 45
        H -= 1
        if (H < 0) {
            H = 24 + H
        }
    } else {
        M -= 45
    }

    println("$H $M")
}
